cpp/tests/unit_tests/kernels/routing/routingRenormalizeTest.cpp

```diff
@@ -0,0 +1,217 @@
+                    // static_cast<int>(expIdx[ie].idx));
+                }
+            }
+        }
+    }
+
+    void allocateBuffers(RoutingKernelTestParam const& param) override
+    {
+        RoutingKernelTest<T>::allocateBuffers(param);
+        int64_t scoresSize = param.numTokens * param.numExperts;
+        this->mPtrScoresHost = mBufferManager->pinned(ITensor::makeShape({scoresSize}), TRTDataType<T>::value);
+        this->mPtrScoresDevice = mBufferManager->gpu(ITensor::makeShape({scoresSize}), TRTDataType<T>::value);
+    }
+
+    template <typename RoutingData>
+    void setParams(RoutingKernelTestParam const& param, RoutingData& routingData)
+    {
+        RoutingKernelTest<T>::setCommonParams(param, routingData);
+
+        if (sizeof(T) == 4)
+        {
+            routingData.mDtypeExpW = btg::Dtype::Fp32;
+        }
+        else
+        {
+            routingData.mDtypeExpW = btg::Dtype::Bfloat16;
+        }
+
+        // Special case for RenormalizeNaive
+        routingData.mDoSoftmaxBeforeTopK = param.routingMethod == RoutingMethodType::RenormalizeNaive;
+        routingData.mNormTopkProb = param.routingMethod == RoutingMethodType::RenormalizeNaive;
+
+        routingData.mPtrScores = bufferCast<T>(*this->mPtrScoresDevice);
+        routingData.mPtrExpertIdx = reinterpret_cast<PackedType*>(bufferCast<int8_t>(*this->mPtrExpertIdxDevice));
+    }
+
+    void callTestedFunction(
+        RoutingKernelTestParam const& param, tensorrt_llm::runtime::ITensor::SharedPtr& workspaceDevice) override
+    {
+        moe::dev::routingRenormalize::Data routingData;
+        setParams(param, routingData);
+        moe::dev::routingRenormalize::run(routingData, mStream->get());
+    }
+};
+
+TYPED_TEST_SUITE(RoutingRenormalizeKernelTest, FloatAndBf16Types);
+
+TYPED_TEST(RoutingRenormalizeKernelTest, ClusterLevelParallelization)
+{
+    RoutingKernelTestParam param(RoutingMethodType::Renormalize, /*numTokens=*/10,
+        /*numExperts=*/128, /*topK=*/8,
+        /*expertParallelization=*/1, /*expertParallelizationId=*/0,
+        /*paddingLog2=*/3, /*localExpertsStrideLog2=*/0,
+        /*usePdl=*/true, /*getExpWeights=*/true,
+        /*nGroup*/ 0, /*topkGroup*/ 0, /*routedScalingFactor*/ 1.0f);
+    this->runTest(param);
+};
+
+TYPED_TEST(RoutingRenormalizeKernelTest, ClusterLevelParallelizationWithExpertParallelization)
+{
+    RoutingKernelTestParam param(RoutingMethodType::Renormalize, /*numTokens=*/100,
+        /*numExperts=*/128, /*topK=*/8,
+        /*expertParallelization=*/2, /*expertParallelizationId=*/1,
+        /*paddingLog2=*/3, /*localExpertsStrideLog2=*/0,
+        /*usePdl=*/true, /*getExpWeights=*/true,
+        /*nGroup*/ 0, /*topkGroup*/ 0, /*routedScalingFactor*/ 1.0f);
+    this->runTest(param);
+};
+
+TYPED_TEST(RoutingRenormalizeKernelTest, DeviceLevelParallelization)
+{
+    RoutingKernelTestParam param(RoutingMethodType::Renormalize, /*numTokens=*/10000,
```

Comment:
I don't see any tests for `RenormalizeNaive`, can we have a simple test for that here? It should behave the same as the default `Renormalize` right?